src/ts.c

```diff
@@ -226,9 +250,21 @@ void add_ts_packet_data(unsigned char *buf, mumudvb_ts_packet_t *pkt, int data_l
 		if(pkt->status_partial!=EMPTY)
 			log_message(log_module, MSG_FLOOD, "Unfinished packet and beginning of a new one, we drop the started one len: %d\n", pkt->len_partial);
 		//We copy the data to the partial packet
-		pkt->status_partial=STARTED;
 		pkt->cc=cc;
 		pkt->pid=pid;
+		if(data_left<3)
+		{
+			pkt->status_partial=PARTIAL_HEADER;
+			memcpy(pkt->data_partial,buf,data_left);
+			pkt->len_partial=data_left;
+			pkt->expected_len_partial=0;
+			log_message(log_module, MSG_FLOOD, "Starting a packet with partial length header, PID %d cc %d len %d\n",
+					pkt->pid,
+					pkt->cc,
+					pkt->len_partial);
+			return;
```

Comment:
I would prefer a if() {A} - else() {B} structure rather than a structure if() {A return} B
Clearer to read